Move latest to master

joebochill opened the pull request:
<!-- If this pull request fixes an Issue, link it below. If not, you can remove the line below -->
<!-- Include a bulleted list summarizing the main changes you have made in this PR -->
#### Changes proposed in this Pull Request:
- Move latest code to master

jeffvg commented:
still waiting for android to start but here is iOS

![image](https://user-images.githubusercontent.com/37150565/113338510-35ac6500-92f7-11eb-9cc9-12e453c5d13d.png)

jeffvg commented:
![image](https://user-images.githubusercontent.com/37150565/113339704-e36c4380-92f8-11eb-836c-3705bbf920b2.png)

joebochill commented:
@jeffvg, @daileytj saw the same thing in one of the RN repos yesterday and a fresh clone cleared it up. Are you fresh?

jeffvg commented:
> @jeffvg, @daileytj saw the same thing in one of the RN repos yesterday and a fresh clone cleared it up. Are you fresh?

Yes fresh clone and double checked

daileytj commented:
> > @jeffvg, @daileytj saw the same thing in one of the RN repos yesterday and a fresh clone cleared it up. Are you fresh?
> 
> Yes fresh clone and double checked

Did you try deleting expo from your simulator and reinstalling?

daileytj commented:
Uggh! On a fresh clone I'm getting the font and icon loading issue again as well. Not sure what's happening here.

jeffvg commented:
> > > @jeffvg, @daileytj saw the same thing in one of the RN repos yesterday and a fresh clone cleared it up. Are you fresh?
> > 
> > 
> > Yes fresh clone and double checked
> 
> Did you try deleting expo from your simulator and reinstalling?

Yes, I made it a habit deleting expo and apps for each deployment

joebochill commented:
Which version of the simulator are you guys running on?

daileytj commented:
> Which version of the simulator are you guys running on?

iPhone 12 - iOS 14.1

jeffvg commented:
> > Which version of the simulator are you guys running on?
> 
> iPhone 12 - iOS 14.1

I'm running the same version as well

jeffvg commented:
Unrecognized font family 'OpenSans-SemiBold'

Is it possible we are having issues with { useFonts } from '@use-expo/font'

joebochill commented:
@daileytj @jeffvg I just switched the implementation to use the `expo-font` package instead of `@use-expo/font`, which is what they use in their example. Also updated to make sure all of the dependency versions are compatible with the latest expo version. Let me know if a fresh clone works now.

It's working for me on iPhone 12 with version 14.2 (but it was also working before this change). I'm still installing 14.1 to see if it runs there, but it's a 5 GB download and taking a little while.

joebochill commented:
Working for me on 14.1 as well:
![image](https://user-images.githubusercontent.com/29152776/113714256-360d7d00-96b6-11eb-8c99-646998136033.png)

jeffvg commented:
> Working for me on 14.1 as well:
> ![image](https://user-images.githubusercontent.com/29152776/113714256-360d7d00-96b6-11eb-8c99-646998136033.png)

I'm still in the same boat with Unrecognized font family 'OpenSans-SemiBold' on iOS. Android doesn't complain about the fonts but still has issues with icons.

![image](https://user-images.githubusercontent.com/37150565/113719703-e7fb7800-96bb-11eb-9a24-074fd9799737.png)

joebochill commented:
Not sure what else I can do here. The example is built with all of the latest compatible packages in accordance with the tutorial instructions on Expo's website. It runs on my simulators on a variety of OS versions and devices and it runs for me on my physical device.

Does it work correctly on a physical device for you?

jeffvg commented:
> Not sure what else I can do here. The example is built with all of the latest compatible packages in accordance with the tutorial instructions on Expo's website. It runs on my simulators on a variety of OS versions and devices and it runs for me on my physical device.
> 
> Does it work correctly on a physical device for you?

It doesn't. Just realized in package.json we are using SDK 40.0.X and when I scan QR code to load app it displays SDK version is not valid. Options are SDK 35.0, 36.0 and 38.0

jeffvg commented:
> > Not sure what else I can do here. The example is built with all of the latest compatible packages in accordance with the tutorial instructions on Expo's website. It runs on my simulators on a variety of OS versions and devices and it runs for me on my physical device.
> > Does it work correctly on a physical device for you?
> 
> It doesn't. Just realized in package.json we are using SDK 40.0.X and when I scan QR code to load app it displays SDK version is not valid. Options are SDK 35.0, 36.0 and 38.0

So strange thing. It finally loaded up on real device. But the odd thing now is I have chinese character for menu icon

joebochill commented:
> It doesn't. Just realized in package.json we are using SDK 40.0.X and when I scan QR code to load app it displays SDK version is not valid. Options are SDK 35.0, 36.0 and 38.0

You need to update your version of the Expo Client app.

joebochill commented:
> So strange thing. It finally loaded up on real device. But the odd thing now is I have chinese character for menu icon and all chevrons are chinese characters

Lol that's definitely not right. You don't have Open Sans or the material icons fonts loaded. Android just happens to fail more gracefully than iOS.

Are you sure you have the latest version of Expo on the simulator? Did you try manually uninstalling it and letting it get reinstalled?

emclaug2 commented:
Are there any updates here @jeffvg @daileytj  @joebochill ?

joebochill commented:
I think Jeff is still playing around with it. I have no way to test it out since everything is working fine for me.

@emclaug2 if you want to clone down and see if you can run it to give us a 4th data point, that would be helpful.

jeffvg commented:
> Are there any updates here @jeffvg @daileytj @joebochill ?

Yeah,  let me look at this again

jeffvg commented:
Yeah. I'm at a loss.
* iOS

![image](https://user-images.githubusercontent.com/37150565/114215708-e9cf7080-9933-11eb-9546-318029a9ca93.png)

*dismiss error
![image](https://user-images.githubusercontent.com/37150565/114215781-066ba880-9934-11eb-8c5d-2c8b36014e28.png)

*click chevron icon
![image](https://user-images.githubusercontent.com/37150565/114215852-1edbc300-9934-11eb-8c05-d0c16c3d3a84.png)

jeffvg commented:
This time I did find something new...

Error running `xcrun simctl openurl E184973B-58AD-4EB3-9C56-E3634560B974 exp://127.0.0.1:19000`: An error was encountered processing the command (domain=NSPOSIXErrorDomain, code=60):
The operation couldn’t be completed. Operation timed out
Operation timed out

xcrun exited with non-zero code: 60

daileytj commented:
I'm still getting the same results as Jeff.

joebochill commented:
I updated to the latest version of expo-cli (3.x -> 4.x) and now I am able to reproduce your error. Seems like they mucked something up.

joebochill commented:
@jeffvg @daileytj  Give it a shot now...I think I finally fixed it.

daileytj commented:
> @jeffvg @daileytj Give it a shot now...I think I finally fixed it.

Working fine now. Just have to clean up those linting errors and this should be good to go.

joebochill commented:
https://github.com/pxblue/pxblue-cli/issues/73

jeffvg reviewed (approved):
all good now. looks good
